### ISSUE TITLE
Provide the ability to read initial temperature profile

### ISSUE THREAD

**Issue by lsoucasse** [ExPlanetology/aragog#10]
For now a linear temperature profile with boundaries set up in the config file is used for initialization. We would like the ability to read temperature profile, coming from previous run of the code.

**Comment by timlichtenberg** [comment]
@djbower Can you please give us some pointers to how/where this was done in SPIDER, and how it should be handled at best in Aragog? Similar to this issues here: https://github.com/ExPlanetology/aragog/issues/4.

**Comment by djbower** [comment]
In `parser.py` you'll need to extend the `_InitialConditionParameters` data class to include some kind of flag/switch that determines how to construct the initial condition.  In general, I tried to separate the objects/dataclasses that store parameters from those that operate on parameters. So for example, `_InitialConditionParameters` stores the parameters, but `InitialCondition` in `core.py` contains the processing functionality for the initial condition.  You will see a similar structure for other combinations of parameters and processing blocks.  In `InitialCondition` there is already a `get_linear` method to set a linear temperature profile, but now you will need to add another method that reads/interpolates a read-in file (so create a method `get_from_file`).  In general, I favor performing non-dimensionalisation in the parameter class so this is dealt with immediately.  In which case, I would read in the data and non-dimensionalise and store the interpolation object in the parameter class, but then use the processing class to actually set the temperature by interpolating to the mesh (since the processing classes in general know about other objects in the system, whereas the parameter classes are designed to be self-contained for each parameter block).  You will of course need to allow the construction of the temperature profile to depend on the switch that is read into the parameters class by updating the `__post_init__` of `InitialCondition`.

So in short, there are data storage classes, that deal with storing the parameters from the cfg and perform non-dimensionalisation and basic pre-processing.  And then there are matching processing classes that mostly do stuff and interact with other objects in the system.  Once you start working with the code a bit I think you'll see the pattern.

**Comment by lsoucasse** [comment]
Thanks @djbower for these precisions? Is the grid supposed to change with time? Are we working with a uniform grid with fixed grid size?

**Comment by djbower** [comment]
Yes that's correct, just a uniform grid with a fixed grid size.

**Comment by timlichtenberg** [comment]
I would like to add here that in principle the grid size may change during a PROTEUS run at some point, since we would like to add planetary accretion in the mid-term. The grid remains fixed and uniform, but the number of grid cells may change.

**Comment by lsoucasse** [comment]
OK, it seems to me that the remeshing and interpolation (if needed) will be inside proteus rather than inside aragog. For now I have just implemented the possibility to set the whole initial temperature field for a given resolution.
